src/tasks/SingularityHandler.h

```diff
@@ -0,0 +1,95 @@
+/**
+ * @file SingularityHandler.h
+ * @author William Chong (wmchong@stnaford.edu)
+ * @brief This class handles type 1 and type 2 singularity.
+ *          For type 1, the joint task is blended with the force task in the singular space.
+ *          For type 2, an open-loop torque is applied proportional to the dot product between 
+ *          the force task and the singular direction and blended. The torque direction can be either direction
+ *          since the singular direction can move pi/2 either direction to be orthogonal to the force vector
+ *          since the singular direction only needs to be in-line with the force vector.
+ * @version 0.1
+ * @date 2024-01-13
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+
+#pragma once 
+
+#include "Sai2Model.h"
+#include <Eigen/Dense>
+
+using namespace Eigen;
+namespace Sai2Primitives {
+
+enum SingularityType {
+    NO_SINGULARITY = 0,
+    TYPE_1_SINGULARITY,
+    TYPE_2_SINGULARITY
+};
+
+class SingularityHandler {
+public:
+    SingularityHandler(std::shared_ptr<Sai2Model::Sai2Model> robot,
+                       const MatrixXd& J_posture,
+                       const double& type_1_tol = 0.95);
+    void updateTaskModel(const MatrixXd& full_jacobian,
```

Comment:
Can we pass a shared pointer to the robot model in the constructor, and only pass the projected jacobian in this update function and do all the necessary computations inside ?